_posts/languages/nodejs/2000-01-01-common-errors.md

```diff
@@ -0,0 +1,48 @@
+---
+title: Node.js Common Errors
+nav: Common Errors
+modified_at: 2020-06-12 00:00:00
+tags: nodejs
+index: 3
+---
+
+You may experience an error in your Node.js application that many customers
+faced when first deploying such application on Scalingo. Here is a list of the
+most common error messages.
+
+## devDependencies Also Contain the Build Dependencies
+
+The `devDependencies` section of the package.json file contains both development
+dependencies and build dependencies. By default Scalingo deployments install the
+dependencies from the `dependencies` section of the package.json file. It may
+lead to error messages such as `ng: not found` or `nest: not found`. In such
+situation, you have a couple of solutions:
+
+- Install all `devDependencies` ([doc]({% post_url
+    languages/nodejs/2000-01-01-start %}#install-devdependencies)).
+- Move the `devDependencies` needed for the build into the `dependencies`
+    section of the package.json file.
+
+## Boot Timeout
+
+You may see the following log lines at the end of your application deployment
+logs:
+
+```text
+[...]
+ Build complete, shipping your container...
+ Waiting for your application to boot...
+ !   Error deploying the application
+ !   → Timeout: my-app took more than 60 seconds to boot
+ !   Application Logs:  https://my.scalingo.com/apps/my-app/logs
+ !   Documentation:     http://doc.scalingo.com/deployment/start-error#timeout
+
+To ssh.osc-fr1.scalingo.com:my-app.git
+ ! [remote rejected] master -> master (pre-receive hook declined)
+error: failed to push some refs to 'ssh.osc-fr1.scalingo.com:my-app.git'
+```
+
+Most of the time, this timeout occurs when your application does not bind the
+port declared by the platform in the environment variable `PORT`. You can see
+how to do that in this [Express example]({% post_url
```

Comment:
Same here. I think it would be better to have a code snippet here.
BTW, if I'm not mistaken boot timeout can occur is app is not binding on port env.PORT but also when it does not bind on 0.0.0.0